cabal.project

```diff
@@ -1,3 +1,6 @@
+-- ATTENTION: care must be taken to keep this file in sync with cabal.project.ci. If you add a package here, add it there (and add a package stanza with ghc-options to enable errors in CI at the bottom of that file).
```

Comment:
These notices are important, and a rather unfortunate necessity: We can’t just pass `--ghc-option=-Werror` to `cabal` in CI, because [a long-standing bug in `cabal`](https://github.com/haskell/cabal/issues/3883) causes it to apply `ghc-options` specified on the CLI to _all_ packages instead of only the local ones.

We could perhaps have added `cabal.project.local` files specifying `-Wwarn` for all local packages, but we’d have had to keep those in sync too, and as that file is the target of `cabal configure`, it would potentially introduce a bunch of work to keep spurious changes to those files out of PRs.

On the balance, this seemed a better solution, albeit unfortunately duplicative; hopefully the `ATTENTION` notices in these files will remind us to keep things synced up. (And if not, maybe we can add a CI check that `cabal.project` is a prefix of `cabal.project.ci`.)